spring-cloud-stream-schema/src/main/java/org/springframework/cloud/stream/schema/client/config/SchemaRegistryClientConfiguration.java

```diff
@@ -25,17 +26,26 @@
 
 /**
  * @author Marius Bogoevici
+ * @author Vinicius Carvalho
  */
 @Configuration
 @EnableConfigurationProperties(SchemaRegistryClientProperties.class)
 public class SchemaRegistryClientConfiguration {
 
+	@Autowired
+	private SchemaRegistryClientProperties schemaRegistryClientProperties;
+
 	@Bean
-	public SchemaRegistryClient schemaRegistryClient(SchemaRegistryClientProperties schemaRegistryClientProperties) {
+	public SchemaRegistryClient schemaRegistryClient() {
+		return createDefaultSchemaClient();
+	}
+
+	private DefaultSchemaRegistryClient createDefaultSchemaClient(){
```

Comment:
Yes I questioned whether I wanted to remove it or not, it used to be there when we were wrapping the CachingRegistryClient ourselves. I think we may just move all this to the autoconfiguration class and use `@Conditional` annotations to decide what we should provide